src/serac/physics/materials/parameterized_solid_functional_material.hpp

```diff
@@ -40,8 +40,8 @@ struct ParameterizedLinearIsotropic {
   SERAC_HOST_DEVICE auto operator()(const DispGradType& du_dX, const BulkType& DeltaK, const ShearType& DeltaG) const
   {
     constexpr auto I       = Identity<dim>();
-    auto           K       = K0 + DeltaK;
-    auto           G       = G0 + DeltaG;
+    auto           K       = K0 + get<0>(DeltaK);
```

Comment:
Why is this needed? Wouldn't this break the AD?

Reply:
(If I'm reading it right, it's throwing away the derivative part of the dual number)

Reply:
This is a little bit confusing. `get<>()` is for tuples, `get_gradient()` and `get_value()` are for `dual`. 

The highlighted code change is related to how `SolidFunctional` previously used to strip out the derivatives of parameters before passing them to the material q-functions. An example of this from the addMaterial function (before):

```cpp
        [this, material](auto /*x*/, auto& state, auto displacement, auto acceleration, auto... params) {
          auto du_dX   = get<DERIVATIVE>(displacement);
          auto d2u_dt2 = get<VALUE>(acceleration);
          auto stress  = material(state, du_dX, serac::get<VALUE>(params)...); // <----------
          ....
```

The problem was that in order to reuse the single-physics solvers (thermal and mechanics) in the thermomechanics module, they each need to know about the other primal field as a "parameter" (displacement and temperature, respectively). But, they also need the gradients of those primal fields to work, so we changed `SolidFunctional` to do this instead:

```cpp
        [this, material](auto /*x*/, auto& state, auto displacement, auto acceleration, auto... params) {
          auto du_dX   = get<DERIVATIVE>(displacement);
          auto d2u_dt2 = get<VALUE>(acceleration);
          auto stress  = material(state, du_dX, params...); // <----------
          ....
```

But that means that the previous materials that only use the value of the parameters now have to index into the tuple from the q-function argument explicitly to pull out the value.

Reply:
Ahhh, I missed that this got moved.